js/main.js

```diff
@@ -0,0 +1,172 @@
+
+const canvas = document.getElementById('wordSearchCanvas');
+const context = canvas.getContext('2d');
+
+/** @const {!Array<number>} */
+const LETTER_FREQUENCY = [
+  8.498,  9.989,  12.191, 16.445, 27.607, 29.835, 31.85,   37.944, 45.49,
+  45.643, 46.935, 50.96,  53.365, 60.114, 67.621, 69.550,  69.645, 77.232,
+  83.559, 92.915, 95.673, 96.651, 99.211, 99.361, 101.355, 101.432
+];
+
+function redraw() {
+  let height = canvas.getBoundingClientRect().height;
+  let width = canvas.getBoundingClientRect().width;
+
+  // console.log('Height is: ' + height + ' Width is: ' + width);
+}
+
+/** @record */
+class GameBoard {
+  constructor() {
+    /** @type {number} */
+    this.startPointX;
+    /** @type {number} */
+    this.startPointY;
+    /** @type {number} */
+    this.gameHeight;
+    /** @type {number} */
+    this.gameWidth;
+    /** @type {number} */
+    this.dimensionsOfSquare;
+    /** @type {!Array<!Array<!Square>>} */
+    this.board;
+  }
+}
+
+/** @record */
+class Square {
+  constructor() {
+    /** @type {number} */
+    this.xCoord;
+    /** @type {number} */
+    this.yCoord;
+    /** @type {string} */
+    this.character;
+  }
+}
+
+
+/**
+ * Returns a word seach game board that is created based of the dimensions
+ * of the allocated screen space with each letter being allocated it's
+ * own square within the board.
+ * @param {number} canvasHeight
+ * @param {number} canvasWidth
+ * @returns {!GameBoard}
+ */
+function createGameBoard(canvasHeight, canvasWidth) {
+  let /** @type {!GameBoard} */ gameBoard = {
```

Comment:
It is recommended to use `const` wherever possible. In JS `const` doesn't mean that variable is fully immutable as you might expect. Instead it means that you can't reassign that variable to a new value. Similar to `final` in java. In this case gameBoard is never reassigned even though it's modified. So `const` should be used here.

At least according to google JS practices.

Reply:
Interesting. I did not know this. Updated.